oval_graph/_xml_parser_comments.py

```diff
@@ -47,26 +46,25 @@ def _prepare_definition_comments(self):
             criteria = definition.find('.//oval-definitions:criteria', ns)
             comment_definition['node'].append(
                 self._create_dict_form_criteria(criteria, description.text))
-            definitions.append(comment_definition)
+            definitions[definition.get('id')] = comment_definition
         return definitions
 
     def _recursive_help_fill_comments(self, comments, nodes):
-        out = nodes
-        out['comment'] = comments['comment']
-        for node, comment in zip(out['node'], comments['node']):
+        for node, comment in zip(nodes, comments):
             node['comment'] = comment['comment']
```

Comment:
Try to think about naming here - we have `node`, `comment`, `comment["node"]`, `node["node"]`, and even `comment["comment"]`, which is quite confusing. In graphs, you have nodes and edges (or relations), at the same time you have the data structure with comments, IDs, references etc., and you need to distinguish between them. Could you e.g. try to describe what is the node of a comment?